File: android/src/main/java/com/margelo/nitro/googlecast/converters/GckMediaLiveSeekableRange+toMediaLiveSeekableRange.kt
package com.margelo.nitro.googlecast.converters

import com.margelo.nitro.googlecast.MediaLiveSeekableRange
import com.google.android.gms.cast.MediaLiveSeekableRange as GckMediaLiveSeekableRange

/**
 * Converts a Google Cast [GckMediaLiveSeekableRange] into a generated [MediaLiveSeekableRange]
 * struct.
 *
 * Reverse of `MediaLiveSeekableRange+toGckMediaLiveSeekableRange.kt`. GCK milliseconds are
 * converted back to the struct's seconds (`Double`).
 */
internal fun GckMediaLiveSeekableRange.toMediaLiveSeekableRange(): MediaLiveSeekableRange =
  MediaLiveSeekableRange(
    startTime = startTime / 1000.0,
    endTime = endTime / 1000.0,
    isMovingWindow = isMovingWindow,
    isLiveDone = isLiveDone
  )

File: ios/NitroGoogleCast/Converters/GCKMediaLiveSeekableRange+toMediaLiveSeekableRange.swift
import Foundation
import GoogleCast

/// Converts a `GCKMediaLiveSeekableRange` into a generated `MediaLiveSeekableRange` struct.
///
/// `GCKMediaLiveSeekableRange` is a receive-only GCK type (no public initializer); this
/// GCK→struct direction is the one used in the real app. All fields are non-optional.
extension GCKMediaLiveSeekableRange {
  func toMediaLiveSeekableRange() -> MediaLiveSeekableRange {
    return MediaLiveSeekableRange(
      startTime: startTime,
      endTime: endTime,
      isMovingWindow: isMovingWindow,
      isLiveDone: isLiveDone
    )
  }
}
